### PR TITLE
MetadataReader: add remarks and example

### PR DESCRIPTION
**Pull request by MSDN-WhiteKnight**
Add remarks and example for [System.Reflection.Metadata.MetadataReader](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.metadata.metadatareader?view=netcore-3.1)

### CODE REVIEW

**Comment by BillWagner** [comment]
I'm happy with these changes @MSDN-WhiteKnight 

I'll :shipit: once @tmat agrees.

**Comment by carlossanlop** [comment]
@tmat can you please review this PR?

File: samples/snippets/csharp/api/system.reflection.metadata/metadatareader/MetadataReaderSnippets.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+using System.Reflection.PortableExecutable;
+
+namespace MetadataReaderSnippets
+{
+    class MetadataReaderSnippets
+    {
+        public static void Run()
+        {
+            //<SnippetMetadataReader>
+            FileStream fs = new FileStream("Example.dll", FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
```
Comment (krwq) [inline samples/snippets/csharp/api/system.reflection.metadata/metadatareader/MetadataReaderSnippets.cs:14]: this could be directly in the using

Comment (MSDN-WhiteKnight) [reply samples/snippets/csharp/api/system.reflection.metadata/metadatareader/MetadataReaderSnippets.cs]: Good point. Moved into using in both cases.

File: samples/snippets/csharp/api/system.reflection.metadata/metadatareader/MetadataReaderSnippets.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                    foreach (TypeDefinitionHandle tdefh in mr.TypeDefinitions)
+                    {
+                        TypeDefinition tdef = mr.GetTypeDefinition(tdefh);
+
+                        string ns = mr.GetString(tdef.Namespace);
+                        string name = mr.GetString(tdef.Name);
+                        Console.WriteLine($"{ns}.{name}");
+                    }
+                }
+            }
```
Comment (Thraka) [inline samples/snippets/csharp/api/system.reflection.metadata/metadatareader/MetadataReaderSnippets.cs:29]: Should it be using the newer `using` style?

```suggestion
            using var fs = new FileStream("Example.dll", FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
            using var peReader = new PEReader(fs);

            MetadataReader mr = peReader.GetMetadataReader();

            foreach (TypeDefinitionHandle tdefh in mr.TypeDefinitions)
            {
                TypeDefinition tdef = mr.GetTypeDefinition(tdefh);

                string ns = mr.GetString(tdef.Namespace);
                string name = mr.GetString(tdef.Name);
                Console.WriteLine($"{ns}.{name}");
            }
```

**Comment by gewarren** [comment]
@tmat Looks like we are still waiting for your review on this one.

**Pull request merged**